apps/website-25/src/components/courses/exercises/FreeTextResponse.tsx

```diff
@@ -1,36 +1,55 @@
 import clsx from 'clsx';
-import CTALinkOrButton from '@bluedot/ui/src/CTALinkOrButton';
-import React from 'react';
+import { CTALinkOrButton } from '@bluedot/ui';
+import React, { useCallback, useEffect } from 'react';
 import ReactMarkdown from 'react-markdown';
+import { useForm } from 'react-hook-form';
 
 type FreeTextResponseProps = {
   // Required
-  title: string;
-  description: string;
   className?: string;
+  description: string;
+  onExerciseSubmit: (exerciseResponse: string) => void;
+  title: string;
+  // Optional
+  exerciseResponse?: string;
+};
+
+type FormData = {
+  answer: string;
 };
 
 const FreeTextResponse: React.FC<FreeTextResponseProps> = ({
-  title,
-  description,
   className,
+  description,
+  exerciseResponse,
+  onExerciseSubmit,
+  title,
 }) => {
-  const [submittedOption, setSubmittedOption] = React.useState<string | null>(null);
-  const [isSaved, setIsSaved] = React.useState<boolean>(false);
+  const [isEditing, setIsEditing] = React.useState<boolean>(false);
+  const { register, handleSubmit, setValue } = useForm<FormData>({
+    defaultValues: {
+      answer: exerciseResponse || '',
+    },
+  });
 
-  const handleAnswerSubmit = () => {
-    setSubmittedOption('test');
-    setIsSaved(true);
-  };
+  useEffect(() => {
```

Comment:
This logic will make one of my comments above a problem (about saving on slow networks).

I'm guessing you've put this here because the exerciseResponse isn't loaded at first render, as it's fetched from the network. I think maybe we'd prefer displaying a loader in Exercise.tsx if either the exercise or the response isn't loaded? So on first render the exerciseResponse is loaded.